### ISSUE TITLE
Issue with Xcode 15: "ld: Assertion failed"

### ISSUE THREAD

**Issue by shanpengli** [fxcoudert/gfortran-for-macOS#60]
The current gFortran does not support MacOS Sonoma

**Comment by adamjstewart** [comment]
I just tried building from source on macOS Sonoma but it failed with:
```
1  0x100afffac  ld::AtomPlacement::findAtom(unsigned char, unsigned long long, ld::AtomPlacement::AtomLoc const*&, long long&) const + 1204
2  0x100b15924  ld::InputFiles::SliceParser::parseObjectFile(mach_o::Header const*) const + 15164
3  0x100b22e30  ld::InputFiles::parseAllFiles(void (ld::AtomFile const*) block_pointer)::$_7::operator()(unsigned long, ld::FileInfo const&) const + 420
4  0x18b159950  _dispatch_client_callout2 + 20
5  0x18b16cba0  _dispatch_apply_invoke + 176
6  0x18b159910  _dispatch_client_callout + 20
7  0x18b16b3cc  _dispatch_root_queue_drain + 864
8  0x18b16ba04  _dispatch_worker_thread2 + 156
9  0x18b3030d8  _pthread_wqthread + 228
ld: Assertion failed: (resultIndex < sectData.atoms.size()), function findAtom, file Relocations.cpp, line 1336.
collect2: error: ld returned 1 exit status
make[5]: *** [Makefile:762: libstdc++.la] Error 1
make[5]: Leaving directory '/private/var/folders/jv/cgkfvslj6nq1l7cw0c8c_8gm0000gn/T/Adam/spack-stage/spack-stage-gcc-13.2.0-cif4pnk3frkkk7ycwb2xfeui2iosa5ad/spack-src/spack-build/aarch64-apple-darwin23.0.0/libstdc++-v3/src'
make[4]: *** [Makefile:795: all-recursive] Error 1
make[4]: Leaving directory '/private/var/folders/jv/cgkfvslj6nq1l7cw0c8c_8gm0000gn/T/Adam/spack-stage/spack-stage-gcc-13.2.0-cif4pnk3frkkk7ycwb2xfeui2iosa5ad/spack-src/spack-build/aarch64-apple-darwin23.0.0/libstdc++-v3/src'
make[3]: *** [Makefile:576: all-recursive] Error 1
make[3]: Leaving directory '/private/var/folders/jv/cgkfvslj6nq1l7cw0c8c_8gm0000gn/T/Adam/spack-stage/spack-stage-gcc-13.2.0-cif4pnk3frkkk7ycwb2xfeui2iosa5ad/spack-src/spack-build/aarch64-apple-darwin23.0.0/libstdc++-v3'
make[2]: *** [Makefile:501: all] Error 2
make[2]: Leaving directory '/private/var/folders/jv/cgkfvslj6nq1l7cw0c8c_8gm0000gn/T/Adam/spack-stage/spack-stage-gcc-13.2.0-cif4pnk3frkkk7ycwb2xfeui2iosa5ad/spack-src/spack-build/aarch64-apple-darwin23.0.0/libstdc++-v3'
make[1]: *** [Makefile:11955: all-target-libstdc++-v3] Error 2
make[1]: *** Waiting for unfinished jobs....
```
Full build log: [spack-build-out.txt](https://github.com/fxcoudert/gfortran-for-macOS/files/12802771/spack-build-out.txt)

@shanpengli can you edit the issue title to be more descriptive?

**Comment by shanpengli** [comment]
Below is the error message:

* installing *source* package ‘frailtypack’ ...
** package ‘frailtypack’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
using C compiler: ‘Apple clang version 15.0.0 (clang-1500.0.40.1)’
sh: /opt/gfortran/bin/gfortran: No such file or directory
using SDK: ‘’
/opt/gfortran/bin/gfortran -arch arm64   -fPIC  -Wall -g -O2  -c  Adonnees.f90 -o Adonnees.o
make: /opt/gfortran/bin/gfortran: No such file or directory

**Comment by fxcoudert** [comment]
This is a bug (crash) in Apple's new linker in Xcode 15. It will be fixed in the Xcode 15.1 version, due to be released soon. A possible work-around is to pass `-Wl,-ld_classic` to gcc or gfortran. Or to pass `-ld_classic` directly to the linker.
